src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/RazorConfiguration.cs

```diff
@@ -12,7 +13,8 @@ public sealed record class RazorConfiguration(
     string ConfigurationName,
     ImmutableArray<RazorExtension> Extensions,
     LanguageServerFlags? LanguageServerFlags = null,
-    bool UseConsolidatedMvcViews = false)
+    bool UseConsolidatedMvcViews = false,
+    bool SuppressAddComponentParameter = false)
```

Comment:
We are adding a new RazorConfiguration option, but it's not used anywhere in the compiler as far as I can see. That feels confusing. I don't remember how exactly is this all wired up, but maybe we can remove the `ConfigureRazorCodeGenerationOptions.SuppressAddComponentParameter` option and use this new one in the compiler as well? Or perhaps this should be part of some tooling-specific config instead of RazorConfiguration?

Reply:
For example, it looks like this could be part of RazorProjectInfo instead of RazorConfiguration.

Reply:
It sets the option in src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/LspProjectEngineFactoryProvider.cs

Reply:
I like the idea of putting it on RazorProjectInfo, but whilst that file isn't the compiler, it is implementing a compiler API in the project engine factory. Having it on the project info means we'd only be able to set it in the configure delegate, and that always feels really flaky to me. We have bugs in the past where some bit was left off one delegate, but included in another.

Then again if it wasn't on RazorConfiguration that would definitely solve the question about the default value it should have in each predefined configuration.

Reply:
Alternatively, we could keep it on `RazorConfiguration` but update the compiler / source generator to use it too instead of the mechanism it currently uses.

Reply:
I just pushed a commit (after checking with @ryzngard) that re-wires the compiler side to use the Configuration option they added instead. We were only configuring it in two places, one of which was already just copying it from the configuration anyway, so it's now one less thing to be get wrong.